src/wasm_worker.js

```diff
@@ -14,7 +14,13 @@ if (ENVIRONMENT_IS_NODE) {
 
   var parentPort = nodeWorkerThreads.parentPort;
 
-  parentPort.on('message', (data) => typeof onmessage === "function" && onmessage({ data: data }));
+  parentPort.on('message', (msg) => global.onmessage?.({ data: msg }));
+
+  var wrappedHandlers = {}
+  function wrapMsgHandler(h) {
+    wrappedHandlers[h] ??= (msg) => h({data: msg});
+    return wrappedHandlers[h];
+  }
```

Comment:
What is this function doing? I can't figure it out.

And a specific question, `wrappedHandlers` is an object, and `h` is a function, so does `wrappedHandlers[h]` not stringify the function to generate a key for the object..? Here is what I mean:
```js
> x = {}
{}
> foo = function foo(x) { return x + 1 }
[Function: foo]
> x[foo] = 10
10
> x
{ 'function foo(x) { return x + 1 }': 10 }
```

Reply:
I'm trying to keep a cache of the functions that have been wrapped.

This is because the function I pass to `parentPort.on` has to match the function I pass to `parentPort.off`.   If we create a new wrapper each time then they will never match and `parentPort.off` will not work so `removeEventListener` won't work either.

I guess I can't use an object though and I should instead use a Set?

Reply:
A WeakMap might be best here. That could associate a wrapper with a function, without keeping the function alive.

Reply:
Done